squad_client/commands/download_attachments.py

```diff
@@ -0,0 +1,86 @@
+from squad_client.shortcuts import get_build
+
+
+logger = logging.getLogger(__name__)
+
+
+class DownloadAttachmentsCommand(SquadClientCommand):
+    command = "download_attachments"
+    help_text = "download the attachments from a SQUAD testrun"
+
+    def register(self, subparser):
+        parser = super(DownloadAttachmentsCommand, self).register(subparser)
+        parser.add_argument(
+            "--group",
+            help="SQUAD group",
+            required=True
+        )
+        parser.add_argument(
+            "--project",
+            help="SQUAD project",
+            required=True
+        )
+        parser.add_argument(
+            "--build",
+            help="Build version. Examples: my-build-version. Or pre-defined build aliases: latest and latest-finished",
+            required=True,
+        )
+        parser.add_argument(
+            "--testrun",
+            help="The SQUAD ID of the testrun",
+            type=int,
+            required=True,
+        )
+        parser.add_argument(
+            "--filenames",
+            nargs="+",
+            help="If not all files are required, specify the names of the files to download",
+        )
+        parser.add_argument(
+            "--debug",
+            action='store_true',
+            help="Set debug mode"
+        )
+
+    def run(self, args):
+        if args.debug:
+            logger.setLevel(logging.DEBUG)
+
+        # Check group, project, testrun
+        group = Squad().group(args.group)
+        if group is None:
+            logger.error(f"Group \"{args.group}\"not found")
+            return False
+
+        project = group.project(args.project)
+        if project is None:
+            logger.error(f"Project \"{group.slug}/{args.project}\" not found")
+            return False
+
+        build = get_build(args.build, project)
+        if build is None:
+            logger.error(f"Build \"{group.slug}/{project.slug}/{args.build}\" not found")
+            return False
+
+        testrun = TestRun(args.testrun)
+        if testrun is None:
+            logger.error(f"TestRun \"{group.slug}/{project.slug}/{args.build}/{args.testrun}\" not found")
+
+        # Check if requested files exist
+        attachment_filenames = [attachment.filename for attachment in testrun.attachments]
+        if args.filenames:
+            for filename in args.filenames:
+                if filename not in attachment_filenames:
+                    logger.error(f"Attachment filename {filename} not found in TestRun \"{group.slug}/{project.slug}/{args.build}/{args.testrun}\". "
+                                 f"Attachments in TestRun are: {attachment_filenames}")
+
+        for attachment in testrun.attachments:
+            if not args.filenames or attachment.filename in args.filenames:
+                logger.debug(f"Downloading {attachment.filename}")
+                with open(attachment.filename, 'wb') as fp:
+                    fp.write(attachment.read())
```

Comment:
I think you can move this snippet to within `shortcuts.py` under a function named like `download_testrun_attachments(testrun_id, filenames)`. It will be easier to get your unit tests done.

Reply:
Thanks for confirming that it makes sense to create a shortcut for this. I've created the shortcut and the unit tests for the shortcut 🙌